infrastructure/testdriver/set_permission.https.html

```diff
@@ -1,17 +1,23 @@
 <!DOCTYPE html>
-<meta charset="utf-8">
+<meta charset="utf-8" />
 <title>TestDriver set_permission method</title>
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <script src="/resources/testdriver.js"></script>
 <script src="/resources/testdriver-vendor.js"></script>
 
 <script>
-promise_test(t => {
-  return test_driver.set_permission({name: "ambient-light-sensor"}, "granted", true);
-}, "Grant Permission for one realm");
+    const descriptor = { name: "geolocation" };
 
-promise_test(t => {
-  return test_driver.set_permission({name: "push", userVisibleOnly: true}, "denied");
-}, "Deny Permission, omit one realm");
+    promise_test(async (t) => {
+        await test_driver.set_permission(descriptor, "granted");
+        permission = await navigator.permissions.query(descriptor);
+        assert_equals(permission.state, "granted");
+    }, "Grant Permission");
+
+    promise_test(async (t) => {
+        await test_driver.set_permission(descriptor, "denied");
+        permission = await navigator.permissions.query(descriptor);
+        assert_equals(permission.state, "denied");
+    }, "Deny Permission");
```

Comment:
Should we really be changing this entire test as part of this?

Reply:
In this case, yes. It doesn’t do any harm to test that setting the permission worked.